### ISSUE TITLE
I suggest adding `livenessProbe` configuration to these two components

### ISSUE THREAD

**Issue by llaoj** [istio/istio#29773]
Both `istiod` and `isti-ingressgateway` are deployed on the same machine as the business pod, and they are both single-instance. Because my business pod does not have resource restrictions, it suddenly occupies very high IO, probably because of memory overflow, leading to `istiod` and `istio-ingressgateway` has stopped serving. Furthermore, all the services exposed through the `istio-ingressgateway` are inaccessible. This is very scary. However, through `kubectl get pods -n istio-system`, the status is displayed as running. The restart pod mechanism of k8s does not take effect .
Probably because the `livenessProbe` is not set for `istiod` and `istio-ingressgateway`. Pod cannot be restarted. So `istiod` and `istio-ingressgateway` cannot restart to restore service. This is a serious problem for me.
Therefore, I suggest adding `livenessProbe` configuration to these two components.
---
`istiod` 和 `istio-ingressgateway` 都和业务pod部署在一台机器上, 他们都是单实例. 因为我的业务pod没有做资源限制, 突然占用了非常高IO,  大概是因为内存溢出, 导致`istiod`和 `istio-ingressgateway` 都停止服务了. 进而, 所有通过`istio-ingressgateway`暴露的服务都无法访问. 这非常可怕.  但是, 通过`kubectl get pods -n istio-system` 查看, 显示状态是 running. k8s的重启pod机制并没有生效.
大概是因为没有给`istiod`和`istio-ingressgateway`设置`livenessProbe`. 导致无法重启pod. 所以`istiod`和`istio-ingressgateway`没有办法重启以恢复服务. 这对我来说是一个严重的问题.
所以, 我建议对这两个组件增加`livenessProbe`配置.

**Comment by hzxuzhonghu** [comment]
Could restart resolve the case here, since the load is readlly high?  This scenario is caused by application interference. Instead, you should limit the resources of the app or keep a separate node for istio components.

**Comment by llaoj** [comment]
@hzxuzhonghu thx, restarting can solve it, you are correct. But it seems necessary to configure a default livenessProbe for isito.

**Comment by hzxuzhonghu** [comment]
liveness probe is necessary to detect whether the components are stuck. It will help fast failure recovery.

**Comment by llaoj** [comment]
Yes, I have added it. But it is not on the default deployment list of istio. I suggest adding it.

**Comment by howardjohn** [comment]
This has been discussed many times in the past if you search github issues (or google, there are lots of discussions outside of Istio). my opinion is making it a default liveness probe is a bad idea, as if the probe has false positives the results can be catastrophic. Users are free to add their own at their own risk

**Comment by kubealex** [comment]
This is actually BLOCKING for environments where policies are enforced (OPA GK for instance) where workloads MUST have readiness/liveness probes defined.

**Comment by howardjohn** [comment]
@kubealex if you need to add a probe to satisfy some policy (which seems suspicious to me - https://srcco.de/posts/kubernetes-liveness-probes-are-dangerous.html for example suggests this may not be a great policy), you are free to add one in your deployment. We will not make changes to the default installation to satisfy policies though, and instead only make them if we feel its nearly universally a good idea, which today we have no evidence to suggest.

**Comment by kubealex** [comment]
> @kubealex if you need to add a probe to satisfy some policy (which seems suspicious to me - https://srcco.de/posts/kubernetes-liveness-probes-are-dangerous.html for example suggests this may not be a great policy), you are free to add one in your deployment. We will not make changes to the default installation to satisfy policies though, and instead only make them if we feel its nearly universally a good idea, which today we have no evidence to suggest.

Actually, that's what I wanted to do, editing the profile of the operator, but livenessProbes are not even taken into consideration when it comes to istio-ingressgateway or istiod setup in the operator CRD. (only readiness)

The idea was not to change the default installation, just adding the possibility in CRD for users to define them, leaving them disabled by default in standard profiles.

**Comment by howardjohn** [comment]
:+1: to making it configurable, I thought this was already possible today

**Comment by kubealex** [comment]
> 👍 to making it configurable, I thought this was already possible today

Unfortunately not!

**Comment by kubealex** [comment]
any news here?
